[Wu XueTing] iP

snowstopxt opened the pull request:
### Snowy
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Snowy frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the ``main`` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

snowstopxt committed:
Level 0

snowstopxt committed:
Add Scanner to read user Input

snowstopxt committed:
Added logic for chat bot and end

snowstopxt committed:
add chats array and list command

snowstopxt committed:
Add Task class, mark and unmark tasks

snowstopxt committed:
added ToDo,Deadline and Event Classes

snowstopxt committed:
Add command for todo

snowstopxt committed:
Add deadline command and check for description + deadline

snowstopxt committed:
added event command, and checks for empty fields

snowstopxt committed:
add delete task

snowstopxt committed:
Add automated testing, small adjustments to exceptions

snowstopxt committed:
more test inputs

snowstopxt committed:
Add write to file and delete record in file features

New methods called writeToFile() and deleteFromFile()

Let's,
* add a write to file feature that updates from the taskList
* delete records in file feature after delete command

snowstopxt committed:
Merge pull request #1 from snowstopxt/branch-Level-7

Add write to file and delete record in file features

snowstopxt committed:
Add data feature in chatbot

Able to recognise dates /from and /to for event

Let's
* recognise date as object instead of string

snowstopxt committed:
Merge pull request #2 from snowstopxt/branch-Level-8

Add data feature in chatbot

snowstopxt committed:
Add minimal OOP classes, common, parser, storage, tasklist, ui

Create new classes for related functions, parser, storage, commands,
tasklist, ui and adapted Snowy main to use the classes.

Let's
* use more OOP in snowy bot
* add classes for related functions

snowstopxt committed:
Merge pull request #3 from snowstopxt/OOP

Add minimal OOP classes, common, parser, storage, tasklist, ui

snowstopxt committed:
Merge pull request #4 from snowstopxt/add-gradle-support

Add gradle support

snowstopxt committed:
Change java home to jdk 17

Gradle supports older version of java, java 17.

miuna9890 commented on src/main/java/snowy/ui/TextUi.java:
```diff
@@ -0,0 +1,60 @@
+import java.io.PrintStream;
+import java.util.Scanner;
+
+public class TextUi {
```
Really well done on the naming convention. Every method has been named well !!

miuna9890 commented on src/main/java/snowy/storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+
+/**
+ * Handles saving and loading of tasks from a file.
+ */
```
header comment seems too vague for the amount of code the class handles. Could include a more detailed header

miuna9890 commented on src/main/java/snowy/data/SnowyException.java:
```diff
@@ -0,0 +1,7 @@
+package snowy.data;
```
Would be better if the package is named exceptions for easier understandability of code organization

miuna9890 commented on src/main/java/snowy/data/SnowyException.java:
```diff
@@ -0,0 +1,7 @@
+package snowy.data;
+
+public class SnowyException extends Exception {
```
Seems like snowy exception is reused for all different exceptions that arise from different input commands. It would be better if different types of exceptions are declared such as DeadlineDateMissingException.

miuna9890 commented on src/main/java/snowy/parser/Parser.java:
```diff
@@ -0,0 +1,131 @@
+        }
+    }
+
+    private Command parseTodoCommand(String arg) throws SnowyException {
```
Would be good to add in header comments for better readability of code

miuna9890 commented on src/main/java/snowy/storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+
+    private final Path taskFilePath;
+
+    public Storage(String directoryPath, String fileName) {
```
Header comment could be included for all methods for better readability

snowstopxt committed:
Add JUnit test cases for Storage and TodoCommand

JUnit test cases for Storage class methods and TodoCommand method.

Test creates directory and file, write to file, delete from file, load
from file, and exception thrown.
Test for valid description and execute todoCommand to add to list.

snowstopxt committed:
Add header comments for most classes

Add header comments for most classes, in parser, tasklist, common.

snowstopxt committed:
Remove unnecessary comments

Remove unecessary header comments for getters, and constructors.

Methods with removed header comments: getSize(), getDescription(),
getTask(), getStatusIcon().

snowstopxt committed:
Change code to meet coding standards

Change the code to adhere to coding standards.
Remove all header comments to avoid merge conflict with javadoc branch.

snowstopxt committed:
Add find command function

Add FindCommand class in common, and parseFindCommand method in Parser
class. New pattern for find arguments.

snowstopxt committed:
Merge branch 'branch-A-CodingStandard'

Fix merge conflict between branch-A-JavaDoc and branch-A-CodingStandard

snowstopxt committed:
Merge branch 'branch-Level-9'

Fix merge conflicts in Parser file

seantham21 commented on src/main/java/snowy/Snowy.java:
```diff
@@ -0,0 +1,62 @@
+import snowy.parser.Parser;
+
+public class Snowy {
+    private static final String FILE_PATH = "../ip/src/main/data";
+    private static final String FILENAME = "snowy.txt";
```
I like how the names of constant variables appropriately follow the coding standard and how this is consistent throughout your code.

seantham21 commented on src/main/java/snowy/Snowy.java:
```diff
@@ -0,0 +1,62 @@
+package snowy;
+
+import snowy.common.Command;
+import snowy.common.CommandResult;
+import snowy.data.SnowyException;
+import snowy.storage.Storage;
+import snowy.tasklist.TaskList;
+import snowy.ui.TextUi;
+import snowy.parser.Parser;
```
I like how all your imports are specific and are arranged in alphabetical order.

seantham21 commented on src/main/java/snowy/common/ExitCommand.java:
```diff
@@ -0,0 +1,21 @@
+import snowy.data.SnowyException;
+
+public class ExitCommand extends Command {
+    public static final String MESSAGE_EXIT_ACKNOWLEDGEMENT = "See you next time. Goodbye!";
```
Might be a nit on my part, but perhaps naming this constant EXIT_MESSAGE would make the name shorter and easier to reference in the future?

seantham21 commented on src/main/java/snowy/common/FindCommand.java:
```diff
@@ -0,0 +1,55 @@
+     * @param keyword for searching
+     * @return list of tasks found
+     */
+    private List<Task> getTasksWithKeywordInDescription(String keyword) {
```
I like that the method names are specific, but perhaps they can be a bit long and can be shortened?

seantham21 commented on src/main/java/snowy/storage/Storage.java:
```diff
@@ -0,0 +1,122 @@
+import snowy.tasklist.Task;
+import snowy.data.SnowyException;
+
+import java.io.*;
```
do you think that perhaps more specific imports can be used here?

seantham21 reviewed (commented):
Overall, I found your code very easy to read and follow, with only minor nits and changes requested surrounding certain naming conventions.
LGTM so good job!

snowstopxt committed:
Add GUI to chatbot

Add DialogBox, MainWindow, Launcher and Main class.
Add new dependencies in build.gradle

snowstopxt committed:
Merge pull request #5 from snowstopxt/branch-level-10

Add GUI to chatbot

snowstopxt committed:
Add assertions to methods

Affected files taskList: toggletask, delete task displayList

snowstopxt committed:
Merge pull request #7 from snowstopxt/branch-A-Assertions

Add assertions to methods

snowstopxt committed:
Improve in code quality

Use better naming conventions, clear up code.

snowstopxt committed:
Merge branch 'branch-A-CodeQuality'

snowstopxt committed:
Add taggin extension

Add new tagCommand, parser for tag and adjusted Task methods

snowstopxt committed:
Add TaskDecoder to save files locally

TaskDecoder class added, and loadStorage method in taskList.
- decodes lines from snowy.txt
- loads tasks into taskList when opening application

snowstopxt committed:
Add UI image

Add UI image to docs folder

To show the UI of snowy chatbot

snowstopxt committed:
Rename UI.png to Ui.png

Wrong name 

Renaming it to Ui

snowstopxt committed:
Fix event time bug

Fix to string for event task to show correct date

snowstopxt committed:
Merge branch 'master' of https://github.com/snowstopxt/ip

snowstopxt committed:
Update README.md

Add user guide for Snowy chatbot

Edited readme.md

snowstopxt committed:
Reset data file

delete data in snowy.txt
